Card/src/main/java/com/paypal/android/card/CardClient.kt

```diff
@@ -1,42 +1,119 @@
+import com.paypal.android.card.api.CardAPI
+import com.paypal.android.card.api.GetOrderRequest
+import com.paypal.android.card.model.CardResult
 import com.paypal.android.core.API
 import com.paypal.android.core.CoreConfig
 import com.paypal.android.core.PayPalSDKError
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 
 /**
  * Use this client to approve an order with a [Card].
  */
-class CardClient internal constructor(private val cardAPI: CardAPI) {
+class CardClient internal constructor(
+    activity: FragmentActivity,
+    private val cardAPI: CardAPI,
+    private val browserSwitchClient: BrowserSwitchClient,
+    private val dispatcher: CoroutineDispatcher
+) {
 
-    constructor(configuration: CoreConfig) :
-            this(CardAPI(API(configuration)))
+    var approveOrderListener: ApproveOrderListener? = null
+
+    private val lifeCycleObserver = CardLifeCycleObserver(this)
 
     /**
-     * Confirm [Card] payment source for an order. Use this method for Kotlin integrations
+     *  CardClient constructor
      *
-     * @param request [CardRequest] for requesting an order approval
+     *  @param [activity] Activity that launches the card client
+     *  @param [configuration] Configuration parameters for client
      */
-    suspend fun approveOrder(request: CardRequest): CardResult =
-        cardAPI.confirmPaymentSource(request.orderID, request.card)
+    constructor(activity: FragmentActivity, configuration: CoreConfig) :
+            this(activity, CardAPI(API(configuration)), BrowserSwitchClient(), Dispatchers.Main)
+
+    init {
+        activity.lifecycle.addObserver(lifeCycleObserver)
+    }
 
     /**
-     * Confirm [Card] payment source for an order. Use this method for Java integrations
+     * Confirm [Card] payment source for an order.
      *
-     * @param request [CardRequest] for requesting an order approval
-     * @param callback callback to get response
+     * @param activity [FragmentActivity] activity used to start 3DS flow (if requested)
+     * @param cardRequest [CardRequest] for requesting an order approval
      */
-    fun approveOrder(request: CardRequest, callback: ApproveOrderCallback) {
-        CoroutineScope(Dispatchers.IO).launch {
+    fun approveOrder(activity: FragmentActivity, cardRequest: CardRequest) {
+        CoroutineScope(dispatcher).launch {
             try {
-                val result = approveOrder(request)
-                callback.success(result)
+                confirmPaymentSource(activity, cardRequest)
             } catch (e: PayPalSDKError) {
-                callback.failure(e)
+                approveOrderListener?.onApproveOrderFailure(e)
+            }
+        }
+    }
+
+    private suspend fun confirmPaymentSource(activity: FragmentActivity, cardRequest: CardRequest) {
+        val response = cardAPI.confirmPaymentSource(cardRequest)
```

Comment:
so are we going to use `Dispatchers.Main` for all our api requests? is it safe to asume that the api request are going to be called in`IO`? perhaps wrapping our requests in `withContext(Dispatchers.IO)`? or was this messing a lot with the unit tests?